### PR TITLE
Run linting + type-checking in PR GitHub Actions

### PR DESCRIPTION
**Pull request by MattIPv4**
## Describe your changes

A thought from #839, this wasn't caught by type-checking because we don't run it -- we should!

Unfortunately, due to https://github.com/vercel/next.js/discussions/47010, we have to rely on a bit of a hack to generate the `next-env.d.ts` file that is needed for type-checking -- we lint an empty directory, which triggers next to generate the required file.

## Notes for testing your change

Types are checked correctly when running `pnpm run types` in either of the packages.

### CODE REVIEW

**Comment by MattIPv4** [comment]
This will need rebasing once #845 is in

File: apps/website/package.json [hunk 1]
```diff
@@ -24,10 +24,15 @@
...
+    "lint:next": "next lint",
+    "lint:prisma:validate": "prisma validate",
+    "lint:prisma:format": "prisma format --check",
+    "lint:prisma": "run-p lint:prisma:*",
+    "lint": "run-p lint:*",
+    "types": "tsc --noEmit",
     "start": "next start",
     "test": "vitest",
-    "format": "prettier . --write && prisma format"
+    "format": "prisma format"
```
Comment (pjeweb) [inline apps/website/package.json:35]: What is the background behind removing prettier here?

I personally use prettier in the editor but having a command sounds good too unless there is a problem with it?

Comment (MattIPv4) [reply apps/website/package.json:35]: `format` in the top-level package runs prettier -- I figured we only needed it in one place (plus the chatbot doesn't define it as `format` either).

If you prefer to have it in here as well, happy to add it back.

Comment (pjeweb) [reply apps/website/package.json:35]: sounds good

**Review by pjeweb — APPROVED** [review]

**Pull request merged**